arch/cortex-m7/src/mpu.rs

```diff
@@ -0,0 +1,702 @@
+//! Implementation of the memory protection unit for the Cortex-M3,
```

Comment:
Can this just be a symlink to the m3 mpu file?

Reply:
Cortex-M7 has 16 MPU regions, while Cortex-M3 has only 8 regions